internal/controller/controllers/agent_controller.go

```diff
@@ -1226,6 +1236,20 @@ func (r *AgentReconciler) getIgnitionToken(ctx context.Context, ignitionEndpoint
 	return string(token), nil
 }
 
+func (r *AgentReconciler) setInfraEnvNameLabel(ctx context.Context, log logrus.FieldLogger, h *common.Host, agent *aiv1beta1.Agent) error {
+	infraEnv, err := r.Installer.GetInfraEnvInternal(ctx, installer.GetInfraEnvParams{InfraEnvID: h.InfraEnvID})
+	if err != nil {
+		return err
+	}
+	if infraEnv.Name == nil {
+		return errors.Errorf("infraEnv %s name is nil", h.InfraEnvID)
+	}
+	if setAgentLabel(log, agent, aiv1beta1.InfraEnvNameLabel, *infraEnv.Name) {
+		return r.updateAndReplaceAgent(ctx, agent)
```

Comment:
Do we need some debug/info log here to note that the agent label was set?

Reply:
:man_shrugging: sure?

Not sure it's such an important thing to log, but I definitely can.

Reply:
Yeah not mandatory, just thought it could help on debug and such.